core/src/main/java/de/danoeh/antennapod/core/export/favorites/FavoritesWriter.java

```diff
@@ -0,0 +1,132 @@
+        Log.d(TAG, "Starting to write document");
+
+        InputStream templateStream = context.getAssets().open("html-export-template.html");
+        String template = IOUtils.toString(templateStream, UTF_8);
+        template = template.replaceAll("\\{TITLE\\}", "Favorites");
+        String[] templateParts = template.split("\\{FEEDS\\}");
+
+        InputStream favTemplateStream = context.getAssets().open(FAVORITE_TEMPLATE);
+        String favTemplate = IOUtils.toString(favTemplateStream, UTF_8);
+
+        InputStream feedTemplateStream = context.getAssets().open(FEED_TEMPLATE);
+        String feedTemplate = IOUtils.toString(feedTemplateStream, UTF_8);
+
+        Map<Long, List<FeedItem>> favoriteByFeed = getFeedMap(getFavorites());
+
+        writer.append(templateParts[0]);
+
+        for (Long feedId : favoriteByFeed.keySet()) {
+            List<FeedItem> favorites = favoriteByFeed.get(feedId);
+            writer.append("<li><div>\n");
+            writeFeed(writer, favorites.get(0).getFeed(), feedTemplate);
+
+            writer.append("<ul>\n");
+            for (FeedItem item : favorites) {
+                writeFavoriteItem(writer, item, favTemplate);
+            }
+            writer.append("</ul></div></li>\n");
+        }
+
+        writer.append(templateParts[1]);
+
+        Log.d(TAG, "Finished writing document");
+    }
+
+    private List<FeedItem> getFavorites() {
+        int page = 0;
+
+        List<FeedItem> favoritesList = new ArrayList<>();
+        List<FeedItem> favoritesPage;
+        do {
+            favoritesPage = DBReader.getFavoriteItemsList(page * PAGE_LIMIT, PAGE_LIMIT);
+            favoritesList.addAll(favoritesPage);
+            ++page;
+        } while (!favoritesPage.isEmpty() && favoritesPage.size() == PAGE_LIMIT);
+
+        // sort in descending order
+        Collections.sort(favoritesList, (lhs, rhs) -> rhs.getPubDate().compareTo(lhs.getPubDate()));
+
+        return favoritesList;
+    }
+
+    /**
+     * Group favorite episodes by feed, sorting them by publishing date in descending order.
+     *
+     * @param favoritesList {@code List} of all favorite episodes.
+     * @return A {@code Map} favorite episodes, keyed by feed ID.
+     */
+    private Map<Long, List<FeedItem>> getFeedMap(List<FeedItem> favoritesList) {
+        Map<Long, List<FeedItem>> feedMap = new TreeMap<>();
+
+        for (FeedItem item : favoritesList) {
+            List<FeedItem> feedEpisodes = feedMap.get(item.getFeedId());
+
+            if (feedEpisodes == null) {
+                feedEpisodes = new ArrayList<>();
+                feedMap.put(item.getFeedId(), feedEpisodes);
+            }
+
+            feedEpisodes.add(item);
+        }
+
+        return feedMap;
+    }
+
+    private void writeFeed(Writer writer, Feed feed, String feedTemplate) throws IOException {
+        String feedInfo = feedTemplate
+                .replaceAll("\\{FEED_IMG\\}", feed.getImageUrl())
```

Comment:
I think `replace` should be enough here. It also replaces all occurrences but does not use a regex

Reply:
done & done :-)